### PR TITLE
👽️ Integrate newly refactored `programDonorSummary` API with the Dashboard (#2021)

### PR DESCRIPTION
**Pull request by mistryrn**
**Description of changes**

* Refactor all instances of `programDonorSummaryEntries` and `programDonorSummaryStats` to use the new combined `programDonorSummary` API endpoint
* Connect "Donor Data Summary" table's "Stats Bar" to the new API to update donor counts based on filters

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

**Comment by mistryrn** [comment]
For reference, the new API changes were created in this ticket: https://github.com/icgc-argo/platform-api/issues/452

We've essentially gone from two separate endpoints, `programDonorSummaryEntries` and `programDonorSummaryStats`, to a single endpoint `programDonorSummary`, which contains an `entries` object and a `stats` object that are equivalent to the results from the previous endpoints.

This allows us to more easily apply filters across both the entries and the stats, and have the results in sync at all times.

File: components/pages/submission-system/program-dashboard/StatsBar/index.tsx [hunk 1]
```diff
@@ -91,29 +91,35 @@ const Statistic: React.ComponentType<{ quantity: String; description: String }>
...
+      program: {
+        commitmentDonors: 0,
+      },
     },
-  );
+    loading,
+  } = useQuery<DashboardSummaryData, DashboardSummaryDataVariables>(DASHBOARD_SUMMARY_QUERY, {
+    variables: { programShortName: programShortName },
+    pollInterval: !pollingTimeout ? POLL_INTERVAL_MILLISECONDS : 0,
+  });
```
Comment (caravinci) [inline components/pages/submission-system/program-dashboard/StatsBar/index.tsx:107]: `prettier` can be a jerk sometimes! eh?

File: components/pages/submission-system/program-dashboard/StatsBar/index.tsx [hunk 2]
```diff
@@ -91,29 +91,35 @@ const Statistic: React.ComponentType<{ quantity: String; description: String }>
...
-    {
-      variables: { programShortName: programShortName },
-      pollInterval: !pollingTimeout ? POLL_INTERVAL_MILLISECONDS : 0,
+  const {
+    data: { programDonorSummary: { stats: programDonorSummaryStats }, program } = {
+      programDonorSummary: {
+        stats: EMPTY_PROGRAM_SUMMARY_STATS,
+      },
+      program: {
+        commitmentDonors: 0,
```
Comment (caravinci) [inline components/pages/submission-system/program-dashboard/StatsBar/index.tsx:100]: nice catch with this missing default!

**Review by caravinci — APPROVED** [review]
🚢 ready to roll out

**Pull request merged**